### PR TITLE
feat: add `@stylexjs/stylex` example

### PR DESCRIPTION
**Pull request by himself65**

### CODE REVIEW

**Review by himself65 — COMMENTED** [review]
First. stylex doesn't support esm module, so I use vite cjs plugin to transform it to esm. But I think in dev mode, server inline the code twice so make the error happened

**Review by himself65 — COMMENTED** [review]
First. stylex doesn't support esm module, so I use vite cjs plugin to transform it to esm. But I think in dev mode, server inline the code twice so make the error happened

**Comment by dai-shi** [comment]
> stylex doesn't support esm module

So, is it essentially the same as #110?

**Comment by himself65** [comment]
> > stylex doesn't support esm module
> 
> So, is it essentially the same as #110?

CJS not supported is the vital issue. They suggest using `vite-plugin-commonjs`. However, doesn't work in waku dev mode.

**Review by himself65 — COMMENTED** [review]
Ok, I investigated a while and I believe is plugin stylex bug with SSR

**Comment by HorusGoul** [comment]
The plugin mostly works (https://github.com/HorusGoul/waku-stylex-demo), but I encountered an issue that prevented me from installing a Waku demo on the mono repo of the plugin to add tests to it.

I believe it has to do with having multiple react versions in the same mono repo, for some reason, Waku is trying to pick the one from the top-level node_modules, see this PR: https://github.com/HorusGoul/vite-plugin-stylex/pull/22

**Comment by himself65** [comment]
> I believe it has to do with having multiple react versions in the same mono repo, for some reason, Waku is trying to pick the one from the top-level node_modules, see this PR: https://github.com/HorusGoul/vite-plugin-stylex/pull/22

/cc @dai-shi any idea? I think this is package manager behavior

**Comment by dai-shi** [comment]
I'm not quite following. So, is it like this PR is working for @himself65 without dual package issue, but @HorusGoul has the dual package issue?

**Comment by dai-shi** [comment]
> Ok, I investigated a while and I believe is plugin stylex bug with SSR

What does it mean? It feels like this example has a limitation, or no?

**Comment by himself65** [comment]
> > Ok, I investigated a while and I believe is plugin stylex bug with SSR
> 
> What does it mean? It feels like this example has a limitation, or no?

Just we have to add `@stylex stylesheet;` in one CSS stylesheet to make things work. Not a limitation

**Comment by himself65** [comment]
> I'm not quite following. So, is it like this PR is working for @himself65 without dual package issue, but @HorusGoul has the dual package issue?

No. dual package issue happens on his repo. Not waku

File: examples/12_css/vite.config.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import { defineConfig } from 'vite';
 import { vanillaExtractPlugin } from '@vanilla-extract/vite-plugin';
+import styleX from 'vite-plugin-stylex';
 
 export default defineConfig({
-  plugins: [vanillaExtractPlugin({ emitCssInSsr: true })],
+  ssr: {
+    external: ['@stylexjs/stylex'],
+  },
```
Comment (dai-shi) [inline examples/12_css/vite.config.ts:8]: Does it work as expected?
Anyway, I would like to merge #289 first. Please be patient.

Comment (himself65) [reply examples/12_css/vite.config.ts:8]: yes

File: examples/12_css/package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   },
   "dependencies": {
     "@hono/node-server": "1.3.3",
+    "@stylexjs/stylex": "^0.3.0",
```
Comment (dai-shi) [inline examples/12_css/package.json:13]: We remove `^` in dependencies in examples.

File: package.json [hunk 3]
```diff
@@ -75,5 +75,10 @@
     "typescript": "^5.3.3",
     "wait-port": "^1.1.0",
     "waku": "workspace:*"
+  },
+  "pnpm": {
+    "overrides": {
+      "vite": "5.0.9"
+    }
```
Comment (dai-shi) [inline package.json:82]: Even with this?

File: examples/12_css/package.json [hunk 4]
```diff
@@ -24,6 +25,7 @@
     "@vanilla-extract/vite-plugin": "3.9.3",
     "@vitejs/plugin-react": "4.2.1",
     "typescript": "5.3.3",
-    "vite": "5.0.9"
+    "vite": "5.0.9",
+    "vite-plugin-stylex": "^0.4.0"
```
[the remark this answers is not in the document]
Comment (himself65) [reply examples/12_css/package.json]: this is my mistake

File: package.json (hunk 3 above)
Comment (himself65) [reply package.json:82]: I think so, some upstream package use different version of vite

**Pull request merged**